src/attached_probe.cpp

```diff
@@ -333,15 +334,20 @@ void AttachedProbe::load_prog()
   if (strrchr(name, ':') != NULL)
     namep = strrchr(name, ':') + 1;
 
+  snprintf(log_buf, sizeof(log_buf), "could not determine kernel version");
```

Comment:
Does this error message show up if we don't ever get to try prog_load? This confused me at first so could either do with some refactoring or a comment to clarify what it's for.

Reply:
yeah, if bcc_prog_load/bpf_prog_load is called at least once it will zero the first byte of the message: https://github.com/iovisor/bcc/blob/master/src/cc/libbpf.c#L509.

Anyway, as you pointed out, it is far from obvious. I'll keep a boolean tracking whether *_prog_load was called at least once and throw exception if not.